external/js/cothority/lib/skipchains/skipchains.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+
+const net = require("../net");
+const protobuf = require("../protobuf");
+const misc = require("../misc");
+
+const skipchainID = "Skipchain";
+
+class SkipchainClient {
+
+    /**
+     * Returns a new skipchain client from a roster
+     *
+     * @param {cothority.Roster} roster the roster over which the client can talk to
+     * @param {string} last know skipblock ID in hexadecimal format
+     * @returns {SkipchainClient} A client that can talks to the skipchain services
+     */
+    constructor(group, roster,lastID) {
+        this.lastRoster = roster;
+        this.lastID = misc.hexToUint8Array(lastID);
+        this.socket = new net.RosterSocket(this.lastRoster);
+        this.group = group;
+    }
+
+    /**
+     * Returns the latest known skipblockID
+     *
+     * @returns {string} hexadecimal encoded skipblockID
+     */
+    get latestID() {
+        return misc.uint8ArrayToHex(this.lastID);
```

Comment:
If there was a separate javascript object for SkipblockID, then you could implement toString on it, and this would be a bit more readable/automatic right?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString